### PR TITLE
Add API to WiFi class to configure wdog feed function

### PR DESCRIPTION
**Pull request by pennam**
This pr add the possibility to configure the watchdog feed function through API in the WiFi class and trigger the watchdog during download process.

(TODO implement timeout) -> this should go to Arduino_IoT_library i think

needs https://github.com/arduino-libraries/Arduino_Portenta_OTA/pull/10
and https://github.com/arduino-libraries/ArduinoIoTCloud/pull/261

### CODE REVIEW

File: libraries/WiFi/src/WiFi.h [hunk 1]
```diff
@@ -49,6 +49,8 @@ extern "C" {
 #define DEFAULT_AP_CHANNEL 6
 #endif
 
+#define WIFI_HAS_FEED_WATCHDOG_FUNC
```
Comment (aentinger) [inline libraries/WiFi/src/WiFi.h:52]: Please use a more unique name, e.g. `ARDUINO_PORTENTA_H7_WIFI_HAS_FEED_WATCHDOG_FUNC`

File: libraries/WiFi/src/WiFi.h [hunk 2]
```diff
@@ -49,6 +49,8 @@ extern "C" {
...
+
 namespace arduino {
 
 typedef void* (*voidPrtFuncPtr)(void);
```
Comment (aentinger) [inline libraries/WiFi/src/WiFi.h:56]:
```C++
typedef void(*ArduinoPortentaH7WiFiFeedWatchdogFuncPtr)(void);
```
also
```diff
-typedef void* (*voidPrtFuncPtr)(void);
+typedef void (*voidPrtFuncPtr)(void);
```

**Review by aentinger — CHANGES_REQUESTED** [review]

File: libraries/WiFi/src/WiFi.h (hunk 1 above)
Comment (pennam) [reply libraries/WiFi/src/WiFi.h]: fixed with https://github.com/arduino/ArduinoCore-mbed/pull/258/commits/28bae3e014097cf9621fc911a795c10a922e2ba7

**Review by aentinger — APPROVED** [review]
Tested - works :heavy_check_mark:

**Pull request merged**